OpenRA.Mods.Common/Traits/Attack/AttackFrontal.cs

```diff
@@ -33,26 +39,34 @@ public AttackFrontal(Actor self, AttackFrontalInfo info)
 			this.info = info;
 		}
 
+		public static bool WithinFacingTolerance(int facing, int desiredFacing, int facingTolerance)
+		{
+			if (facingTolerance == 0 && facing == desiredFacing)
+				return true;
+
+			var delta = Util.NormalizeFacing(desiredFacing - facing);
+			if (delta <= facingTolerance || delta >= 256 - facingTolerance)
+				return true;
+
+			return false;
+		}
+
 		protected override bool CanAttack(Actor self, Target target)
 		{
 			if (!base.CanAttack(self, target))
 				return false;
 
-			var f = facing.Facing;
 			var pos = self.CenterPosition;
 			var targetedPosition = Info.AttackTargetCenter ? target.CenterPosition : target.Positions.PositionClosestTo(pos);
 			var delta = targetedPosition - pos;
-			var facingToTarget = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : f;
-
-			if (Math.Abs(facingToTarget - f) % 256 > info.FacingTolerance)
-				return false;
+			var facingToTarget = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : facing.Facing;
```

Comment:
If `delta.HorizontalLengthSquared == 0` then `facing.Facing` will always be within tolerance with itself.  The following would be simpler and clearer:
```
if (delta.HorizontalLengthSquared == 0)
	return true;

return WithinFacingTolerance(facing.Facing, delta.Yaw.Facing, info.FacingTolerance);
```